apps/nfid-frontend/src/features/identitykit/service/canister-calls-helpers/icrc1-transfer.ts

```diff
@@ -0,0 +1,61 @@
+import { Agent, AnonymousIdentity, HttpAgent } from "@dfinity/agent"
+import { decodeIcrcAccount } from "@dfinity/ledger-icrc"
+
+import { exchangeRateService } from "@nfid/integration"
+
+import { idlFactory as icrc1and2IDL } from "../../idl/token-pepe-ledger"
+import { RPCMessage } from "../../type"
+import { actorService } from "../actor.service"
+import { IC_HOSTNAME } from "../method/interactive/icrc49-call-canister-method.service"
+import { TransferMetadata } from "./interfaces"
+
+export const getIcrc1TransferMetadata = async (
+  message: MessageEvent<RPCMessage>,
+  args: any,
+): Promise<TransferMetadata> => {
+  const delegation = new AnonymousIdentity()
+
+  const agent: Agent = new HttpAgent({
+    host: IC_HOSTNAME,
+    identity: delegation,
+  })
+
+  const actor = actorService.getActor<any>(
+    message.data.params.canisterId,
+    icrc1and2IDL,
+    agent as never,
+  )
+
+  const { owner, subaccount } = decodeIcrcAccount(message.data.params.sender)
+
+  const [balance, symbol, decimals, fee] = await Promise.all([
+    actor.icrc1_balance_of({
```

Comment:
why not icrc1_metadata?

Reply:
makes sense, actually was copied from other methods, but i personally don't like return type of metadata, its [[,symbol], [, decimals], [, [fee]], i am not sure it's required